main/params.go

```diff
@@ -66,252 +67,262 @@ var (
+	fs.String(ipcsPathKey, defaultString, "The directory (Unix) or named pipe name prefix (Windows) for IPC sockets")
 
 	// Router Configuration:
-	fs.DurationVar(&Config.ConsensusGossipFrequency, "consensus-gossip-frequency", 10*time.Second, "Frequency of gossiping accepted frontiers.")
-	fs.DurationVar(&Config.ConsensusShutdownTimeout, "consensus-shutdown-timeout", 5*time.Second, "Timeout before killing an unresponsive chain.")
+	fs.Duration(consensusGossipFrequencyKey, 10*time.Second, "Frequency of gossiping accepted frontiers.")
+	fs.Duration(consensusShutdownTimeoutKey, 5*time.Second, "Timeout before killing an unresponsive chain.")
 
-	fdLimit := fs.Uint64("fd-limit", ulimit.DefaultFDLimit, "Attempts to raise the process file descriptor limit to at least this value.")
+	// File Descriptor Limit
+	fs.Uint64(fdLimitKey, ulimit.DefaultFDLimit, "Attempts to raise the process file descriptor limit to at least this value.")
 
 	// Subnet Whitelist
-	whitelistedSubnets := fs.String("whitelisted-subnets", "", "Whitelist of subnets to validate.")
+	fs.String(whitelistedSubnetsKey, "", "Whitelist of subnets to validate.")
 
-	ferr := fs.Parse(os.Args[1:])
+	return fs
+}
 
-	if *version { // If --version used, print version and exit
-		format := "%s ["
-		args := []interface{}{
-			node.Version,
-		}
+// getViper returns the viper environment from parsing config file from default search paths
+// and any parsed command line flags
+func getViper() (*viper.Viper, error) {
+	v := viper.New()
 
-		{
-			networkID, err := constants.NetworkID(*networkName)
-			if err != nil {
-				Err = err
-				return
-			}
-			networkGeneration := constants.NetworkName(networkID)
-			if networkID == constants.MainnetID {
-				format += "network=%s"
-			} else {
-				format += "network=testnet/%s"
-			}
-			args = append(args, networkGeneration)
-		}
+	fs := avalancheFlagSet()
+	pflag.CommandLine.AddGoFlagSet(fs)
+	pflag.Parse()
+	if err := v.BindPFlags(pflag.CommandLine); err != nil {
+		return nil, err
+	}
 
-		{
-			format += ", database=%s"
-			args = append(args, dbVersion)
-		}
-
-		{
-			if GitCommit != "" {
-				format += ", commit=%s"
-				args = append(args, GitCommit)
-			}
+	if configFile := v.GetString(configFileKey); configFile != defaultString {
+		v.SetConfigFile(configFile)
+		if err := v.ReadInConfig(); err != nil {
+			return nil, err
 		}
+	}
 
-		format += "]\n"
+	return v, nil
+}
 
-		fmt.Printf(format, args...)
-		os.Exit(0)
+// setNodeConfig sets attributes on [Config] based on the values
+// defined in the [viper] environment
+func setNodeConfig(v *viper.Viper) error {
+	// Logging:
+	loggingConfig, err := logging.DefaultConfig()
+	if err != nil {
+		return err
+	}
+	logsDir := v.GetString(logsDirKey)
+	if logsDir != "" {
+		loggingConfig.Directory = logsDir
+	}
+	logFileLevel, err := logging.ToLevel(v.GetString(logLevelKey))
```

Comment:
you can do `loggingConfig.LogLevel, err = logging.ToLevel(v.GetString(logLevelKey))` to save a line and variable allocation

Reply:
Updated